Remotion/ObjectBinding/Web.ClientScript/Scripts/BocBooleanValue.ts

```diff
@@ -0,0 +1,187 @@
+
+    if (newValue == falseValue) {
+        checkedState = 'false';
+        iconSrc = this._falseIconUrl;
+        if (falseDescription == null)
+            description = this._falseDescription;
+        else
+            description = falseDescription;
+    }
+    else if (newValue == nullValue) {
+        checkedState = 'mixed';
+        iconSrc = this._nullIconUrl;
+        if (nullDescription == null)
+            description = this._nullDescription;
+        else
+            description = nullDescription;
+    }
+    else if (newValue == trueValue) {
+        checkedState = 'true';
+        iconSrc = this._trueIconUrl;
+        if (trueDescription == null)
+            description = this._trueDescription;
+        else
+            description = trueDescription;
+    } // TODO: add default case to prevent variables from being undefined
+
+    
+    link.setAttribute('aria-checked', checkedState!);
+    icon.src = iconSrc!;
+    if (label == null)
+    {
+      link.title = description!;
+    }
+    else
+    {
+      label.innerHTML = description!;
+    }
+    $(hiddenField).change();
+  }
+}
+
+//  Initializes the strings used to represent the true, false and null values.
+//  Call this method once in a startup script.
+function BocBooleanValue_InitializeGlobals(
+    key: string,
+    trueValue: string, 
+    falseValue: string, 
+    nullValue: string, 
+    trueDescription: string,
+    falseDescription: string,
+    nullDescription: string,
+    trueIconUrl: string, 
+    falseIconUrl: string, 
+    nullIconUrl: string)
+{
+  _bocBooleanValue_Resources[key] = new BocBooleanValue_Resource(
+      trueValue,
+      falseValue,
+      nullValue,
+      trueDescription,
+      falseDescription,
+      nullDescription,
+      trueIconUrl,
+      falseIconUrl,
+      nullIconUrl);
+}
+
+// Selected the next value of the tri-state checkbox, skipping the null value if isRequired is true.
+// link: The anchor tag representing the clickable area.
+// icon: The icon representing the tri-state checkbox.
+// label: The label containing the description for the value. null for no description.
+// hiddenField: The hidden input field used to store the value between postbacks.
+// isRequired: true to enqable the null value, false to limit the choices to true and false.
+function BocBooleanValue_SelectNextCheckboxValue (
+  key: string,
+  link: HTMLAnchorElement,
+  icon: HTMLImageElement,
+  label: Nullable<HTMLElement>,
+  hiddenField: HTMLInputElement,
+  isRequired: boolean,
+  trueDescription: Nullable<string>,
+  falseDescription: Nullable<string>,
+  nullDescription: Nullable<string>)
+{
+  var resource = _bocBooleanValue_Resources[key]!;
+  resource.SelectNextCheckboxValue(
+  link,
+  icon,
+  label,
+  hiddenField,
+  isRequired,
+  trueDescription,
+  falseDescription,
+  nullDescription);
+}
+
+function BocBooleanValue_OnKeyDown (context: HTMLAnchorElement)
+{
+  // TODO: Better provide the event via parameter instead of ambient usage
```

Comment:
Don't be so mean to senior code. Okay, it's not exactly older than our colleagues, but it's not that far off. Give it a year or two and we might get there...

(please bugfix task)